### PR TITLE
feat: Rust API.

### PR DESCRIPTION
**Pull request by amunra**
* Split Rust code into `questdb-rs` and `questdb-rs-ffi` crates.
* The intention is to publish the first crate, but not the latter.
* Added Rust docs.

### CODE REVIEW

File: questdb-rs/src/ingress/mod.rs [hunk 1]
```diff
@@ -186,6 +126,10 @@ impl <'a> TableName<'a> {
 
         Ok(Self { name: name })
     }
+
+    pub unsafe fn new_unchecked(name: &'a str) -> Self {
```
Comment (eugenels) [inline questdb-rs/src/ingress/mod.rs:130]: This fn signature is a bit misleading. 
Unsafe doesn't make much sense for &str. &str must always be valid UTF-8.

File: questdb-rs-ffi/src/lib.rs [hunk 2]
```diff
@@ -313,10 +314,10 @@ pub struct line_sender_table_name
 }
 
 impl line_sender_table_name {
-    fn as_name<'a>(&self) -> TableName<'a> {
-        let str_name = unsafe { std::str::from_utf8_unchecked(
-            slice::from_raw_parts(self.buf as *const u8, self.len)) };
-        TableName{ name: str_name }
+    unsafe fn as_name<'a>(&self) -> TableName<'a> {
```
Comment (eugenels) [inline questdb-rs-ffi/src/lib.rs:317]: This fn may be safe

File: questdb-rs/src/ingress/mod.rs (hunk 1 above)
Comment (amunra) [reply questdb-rs/src/ingress/mod.rs]: Done.

File: questdb-rs-ffi/src/lib.rs (hunk 2 above)
Comment (amunra) [reply questdb-rs-ffi/src/lib.rs:317]: Done.

**Review by eugenels — APPROVED** [review]

**Pull request merged**